Formula/osmcoastline.rb

```diff
@@ -0,0 +1,25 @@
+class Osmcoastline < Formula
+  desc "Extracts coastline data from OpenStreetMap planet file"
+  homepage "https://osmcode.org/osmcoastline/"
+  url "https://github.com/osmcode/osmcoastline/archive/v2.2.4.tar.gz"
+  sha256 "7a399661b46e4e700b11d6a5163ec7bdc8ad49f0837b524f1f2535cca7b9ee43"
+
+  depends_on "cmake" => :build
+  depends_on "libosmium" => :build
+  depends_on "gdal"
+  depends_on "geos"
+  depends_on "libspatialite"
+
+  uses_from_macos "zlib"
+  uses_from_macos "sqlite"
+
+  def install
+    protozero = Formula["libosmium"].opt_libexec/"include"
+    system "cmake", ".", "-DPROTOZERO_INCLUDE_DIR=#{protozero}", *std_cmake_args
+    system "make", "install"
+  end
+  
+  test do
+#     output = shell_output("#{bin}/runtest.sh")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thank you for your suggestion.
I am going to get advices form the osmcoastline developer.